toolchain/lower/file_context.cpp

```diff
@@ -260,25 +258,29 @@ auto FileContext::CoalesceEquivalentSpecifics() -> void {
         }
       }
     }
+
+    // Update equivalent specific llvm::Functions, once all equivalences are
+    // found and the algorithm has converged on a canonical specific.
```

Comment:
Given that this is now per-generic, and a generic may have multiple canonical specifics, suggesting a slightly different comment:

```suggestion
    // Once all equivalences are found for a generic, update and delete up
    // equivalent specifics.
```